test/e2e/csi_test.go

```diff
@@ -220,4 +220,322 @@ var _ = Describe("Nutanix flavor CSI", Label("capx-feature-test", "csi"), func()
 
 		By("PASSED!")
 	})
+
+	It("Create a cluster with Nutanix CSI 3.0 and use Nutanix Volumes to create PV", Label("csi3"), func() {
+		const flavor = "csi3"
+
+		Expect(namespace).NotTo(BeNil())
+
+		By("Creating a workload cluster")
+		testHelper.deployClusterAndWait(
+			deployClusterParams{
+				clusterName:           clusterName,
+				namespace:             namespace,
+				flavor:                flavor,
+				clusterctlConfigPath:  clusterctlConfigPath,
+				artifactFolder:        artifactFolder,
+				bootstrapClusterProxy: bootstrapClusterProxy,
+			}, clusterResources)
+
+		By("Fetching workload client")
+		workloadProxy := bootstrapClusterProxy.GetWorkloadCluster(ctx, namespace.Name, clusterResources.Cluster.Name)
+		workloadClient = workloadProxy.GetClient()
+
+		By("Checking if CSI namespace exists")
+		csiNamespace := &corev1.Namespace{}
+		csiNamespaceKey := client.ObjectKey{
+			Name: csiNamespaceName,
+		}
+		Eventually(func(g Gomega) {
+			g.Expect(workloadClient.Get(ctx, csiNamespaceKey, csiNamespace)).To(Succeed())
+		}, defaultTimeout, defaultInterval).Should(Succeed())
+
+		By("Checking if CSI deployment exists")
+		csiDeployment := &appsv1.Deployment{}
+		csiDeploymentKey := client.ObjectKey{
+			Name:      csiDeploymentName,
+			Namespace: csiNamespaceName,
+		}
+		Eventually(func(g Gomega) {
+			g.Expect(workloadClient.Get(ctx, csiDeploymentKey, csiDeployment)).To(Succeed())
+		}, defaultTimeout, defaultInterval).Should(Succeed())
+
+		By("Creating CSI Storage class")
+		sc := &storagev1.StorageClass{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "storage.k8s.io/v1",
+				Kind:       "StorageClass",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: csiStorageClassName,
+				Annotations: map[string]string{
+					"storageclass.kubernetes.io/is-default-class": "true",
+				},
+			},
+			Provisioner:          csiProvisioner,
+			ReclaimPolicy:        ptr.To(corev1.PersistentVolumeReclaimDelete),
+			VolumeBindingMode:    ptr.To(storagev1.VolumeBindingWaitForFirstConsumer),
+			AllowVolumeExpansion: ptr.To(false),
+			Parameters: map[string]string{
+				"csi.storage.k8s.io/provisioner-secret-name":            "nutanix-csi-credentials",
+				"csi.storage.k8s.io/provisioner-secret-namespace":       csiNamespaceName,
```

Comment:
What will be the namespace here `csiNamespaceName`.
Last I checked there was still a bug with the controller not coming up correctly when not using the default `ntnx-system`.

Reply:
https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/pull/460/files#diff-4af0f90e90a18ceb4ac3541fbf4ad58be3184faa316ee70fc81f54d06b7caa3fR51
<pre>
csiNamespaceName = "ntnx-system"
</pre>

Reply:
it's `ntnx-system`